docs/template_plugin/src/template_infer_request.cpp

```diff
@@ -266,19 +283,97 @@ void TemplateInferRequest::waitPipeline() {
 void TemplateInferRequest::inferPostprocess() {
     OV_ITT_SCOPED_TASK(itt::domains::TemplatePlugin, _profilingTask[Postprocess]);
     auto start = Time::now();
-    for (auto&& output : _outputs) {
-        auto outputBlob = output.second;
-        auto networkOutput = _networkOutputBlobs[output.first];
+    for (auto&& networkOutput : _networkOutputs) {
+        {
+            // Touch blob to allocate it
+            Blob::Ptr blob;
+            GetBlob(networkOutput.first);
+        }
+        auto outputBlob = _outputs.at(networkOutput.first);
+        auto networkOutputBlob = _networkOutputBlobs[networkOutput.first];
         // perform precision conversion of network output's precision and computational
         // graph output's precision are different
-        if (outputBlob->getTensorDesc().getPrecision() != networkOutput->getTensorDesc().getPrecision()) {
-            blobCopy(networkOutput, outputBlob);
+        //if (outputBlob->getTensorDesc().getPrecision() != networkOutputBlob->getTensorDesc().getPrecision()) {
+        //    blobCopy(networkOutputBlob, outputBlob);
+        //}
+        // TODO: Copy to source blob only if necessary (like commented out statements above but also consider shapes)
+        if (outputBlob->getTensorDesc().getPrecision() != networkOutputBlob->getTensorDesc().getPrecision()) {
+            IE_THROW() << "Disabled any precision conversion for output blobs due to dynamic shape limited implementation";
         }
+        auto tensor = _outputTensors[_executableNetwork->_outputIndex.at(networkOutput.first)];
+        tensor->read(InferenceEngine::as<InferenceEngine::MemoryBlob>(outputBlob)->wmap().as<char*>(), tensor->get_size_in_bytes());
     }
     _durations[Postprocess] = Time::now() - start;
 }
 // ! [infer_request:infer_postprocess]
 
+Blob::Ptr TemplateInferRequest::GetBlob(const std::string& name) {
+    OV_ITT_SCOPED_TASK(itt::domains::TemplatePlugin, "GetBlob");
+    InputInfo::Ptr foundInput;
+    DataPtr foundOutput;
+    Blob::Ptr data;
+    const SizeVector oneVector = { 1 };
+    if (findInputAndOutputBlobByName(name, foundInput, foundOutput)) {
+        // ROI blob is returned only if it was set previously. Otherwise default blob is returned.
+        auto it = _preProcData.find(name);
+        if (it != _preProcData.end()) {
+            data = it->second->getRoiBlob();
+        } else {
+            data = _inputs[name];
+            const auto& dims = m_realShapes.find(name) != m_realShapes.end() ? m_realShapes[name] : foundInput->getTensorDesc().getDims();
```

Comment:
in case of dynamic shapes `foundInput->getTensorDesc().getDims()` is empty I suppose.
And if `m_realShapes` are not set by user, we cannot allocate blobs below in this code.